torch/csrc/stable/tensor.h

```diff
@@ -31,7 +31,13 @@ class Tensor {
   std::shared_ptr<AtenTensorOpaque> ath_;
 
  public:
-  Tensor() = delete;
+  Tensor() {
```

Comment:
Document what this does -- does it create new memory? Who is responsible for cleaning up the memory? (I expect the answer is that this object will clean it up properly). Along that thread, let's also ensure we have tests for memory with this constructor.